test/stdlib/Array_test.rb

```diff
@@ -217,6 +217,8 @@ def test_compact
   def test_concat
     assert_send_type "(Array[Integer], Array[Integer]) -> Array[Integer]",
                      [1,2,3], :concat, [4,5,6], [7,8,9]
+    assert_send_type "(Array[Integer], Array[Integer]) -> self",
+                     Class.new(Array).new, :concat, [4,5,6], [7,8,9]
```

Comment:
I don’t know why even in without changing signature, this added test passed.  But adding it makes better, I think.
Please tell me a way if `rbs` have something to ensure it.  🙏

Reply:
Good point. The test passes because `Class.new(Array).new.is_a?(Array)` holds. The runtime type checker can be improved...

Reply:
😮 Thank you for letting me know the why!